src/time.ts

```diff
@@ -33,3 +33,13 @@ export function getRefreshInterval(isoTime: string): number {
 
   return interval < MAX_INTERVAL ? interval : MAX_INTERVAL;
 }
+
+export function getNearestExpiration(
+  certExp: number,
+  tokenExp: number | undefined
+) {
+  if (tokenExp) {
+    return new Date(Math.min(certExp, tokenExp)).toISOString();
+  }
+  return new Date(certExp).toISOString();
+}
```

Comment:
it would be nice to add some unit tests for this method, should be fairly straightforward now that it's on `src/time.ts` and might help out shaking off any edge cases